xds/internal/xdsclient/transport/transport.go

```diff
@@ -428,13 +453,19 @@ func (t *Transport) recv(stream adsStream) bool {
 	for {
 		resources, url, rVersion, nonce, err := t.recvAggregatedDiscoveryServiceResponse(stream)
 		if err != nil {
-			t.adsStreamErrHandler(err)
+			// If we have already successfully received a message on the stream, then
+			// the OnError callback is called with an error of type IgnoredADSRecvError.
+			if msgReceived {
+				t.onErrorHandler(xdsresource.NewErrorf(xdsresource.ErrorTypeIgnored, err.Error()))
+			} else {
+				t.onErrorHandler(err)
+			}
```

Comment:
This could be shortened as follows:
```golang
if msgReceived {
  err = xdsresource.NewErrorf(xdsresource. ErrTypeStreamFailedAfterRecv, err.Error())
}
t.onErrorHandler(err)
```

Reply:
cool, updated :)